src/canvas/SelectableCanvas.ts

```diff
@@ -667,10 +667,23 @@ export class SelectableCanvas<
    * @return {Boolean}
    */
   isTargetTransparent(target: FabricObject, x: number, y: number): boolean {
+    const ctx = this.contextCache;
+    const retina = this.getRetinaScaling();
+    const size = Math.max(2 * this.targetFindTolerance * retina, 10);
+    if (this.pixelFindCanvasEl.width < size) {
+      // TODO: handle in set
+      this.pixelFindCanvasEl.width = this.pixelFindCanvasEl.height = size;
+    } else {
+      this.clearContext(ctx);
+    }
     // in case the target is the activeObject, we cannot execute this optimization
     // because we need to draw controls too.
-    if (isFabricObjectCached(target) && target !== this._activeObject) {
-      // optimizatio: we can reuse the cache
+    if (
```

Comment:
it happened that i thought a couple of times of this:
```
// in case the target is the activeObject, we cannot execute this optimization
    // because we need to draw controls too.
```
And then i asked myself, why do i have to draw the controls?
The controls can be found with the coordinates.

Reply:
absolutely
I had no idea what was that.
And I think active selection never caches, or at least it didn't in v5. Now I don't remember.